crates/serialize_hierarchy/src/lib.rs

```diff
@@ -262,6 +262,144 @@ impl<T> SerializeHierarchy for Vec<T> {
     }
 }
 
+impl<T: Serialize + DeserializeOwned> SerializeHierarchy for Vector2<T> {
+    fn serialize_path<S>(&self, path: &str) -> Result<S::Serialized, Error<S::Error>>
+    where
+        S: Serializer,
+        S::Error: error::Error,
+    {
+        let idx = ["x", "y"].into_iter().position(|x| x == path);
+        match idx {
+            Some(i) => S::serialize(&self[i]).map_err(Error::SerializationFailed),
+            _ => Err(Error::UnexpectedPathSegment {
+                segment: String::from(path),
+            }),
+        }
```

Comment:
Please don't abbreviate.

```suggestion
        let index = ["x", "y"].into_iter().position(|name| name == path);
        match index {
            Some(index) => S::serialize(&self[index]).map_err(Error::SerializationFailed),
            _ => Err(Error::UnexpectedPathSegment {
                segment: String::from(path),
            }),
        }
```

Please also apply this in the other cases.